### ISSUE TITLE
[Bug] Recursion test fails on appveyor

### ISSUE THREAD

**Issue by chriseth** [ethereum/solidity#6854]
```
fatal error: in "SolidityParser/recursion_depth2": stack overflow
C:\projects\solidity\test\libsolidity\SolidityParser.cpp(610): last checkpoint: "recursion_depth2" test entry
```

**Comment by ekpyron** [comment]
Appveyor failure example:  https://ci.appveyor.com/project/ethereum-free/solidity/builds/24868176
Introducing PR: #6681 
Reverting PR: #6857

**Comment by christianparpart** [comment]
Fixed in #6948.
